packages/x-date-pickers/src/DatePicker/DatePickerToolbar.tsx

```diff
@@ -76,24 +76,12 @@ const DatePickerToolbar = React.forwardRef(function DatePickerToolbar<TDate>(
       return toolbarPlaceholder;
     }
 
-    if (toolbarFormat) {
-      return utils.formatByString(value, toolbarFormat);
-    }
-
-    if (isYearOnlyView(views as DateView[])) {
-      return utils.format(value, 'year');
-    }
-
-    if (isYearAndMonthViews(views as DateView[])) {
```

Comment:
For me this one was a "bug"

```tsx
<MobileDatePicker  defaultValue={dayjs()} views={['year', 'month']}  />
```

![image](https://github.com/mui/mui-x/assets/3309670/54c4dafc-5b3f-471a-8278-7c991695665d)

![image](https://github.com/mui/mui-x/assets/3309670/c7d7294e-f41c-4880-bf6f-a54ad3c8423c)


I would expect to see the year